src/sentry/utils/arroyo.py

```diff
@@ -126,17 +133,54 @@ def initialize_arroyo_main() -> None:
     configure_metrics(metrics_wrapper)
 
 
+class ReusableMultiprocessingPool(ArroyoMultiprocessingPool):
+    """
+    A variant of arroyo's MultiprocessingPool that initializes Sentry for you, and
+    ensures global metric tags in the subprocess are inherited from the main process.
+
+    Also auto-closes the pool when it receives the exit signal.
+
+    Since the pool is only closed in the exit handler, a single instance of this class
+    should be used across all consumer rebalances - that is, it must not be recreated
+    by the strategy factory on every rebalance.
+    """
+
+    def __init__(
+        self,
+        num_processes: int,
+        initializer: Optional[Callable[[], None]] = None,
+    ) -> None:
+        atexit.register(self._shutdown)
```

Comment:
i feel like all this class does is to run this piece of code, but that can also happen inside of `get_reusable_multiprocessing_pool`